pkg/kapp/resources/identified_resources_list_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+package resources
+
+import (
+	"github.com/cppforlife/go-cli-ui/ui"
+	logger2 "github.com/k14s/kapp/pkg/kapp/logger"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func TestIdentifiedResourcesListReturnsLabeledResources(t *testing.T) {
+	mockedResTypes := new(ResourceTypesMock)
+	mockedRess := new(ResourcesMock)
+
+	mockedRess.Mock.On("All").Return(makeNewResources(t), nil)
+
+	ui := ui.NewNoopUI()
+	logger := logger2.NewUILogger(ui)
+	identifiedResources := NewIdentifiedResources(nil, mockedResTypes, mockedRess, []string{}, logger)
+
+	kappLabel := make(map[string]string)
+	kappLabel["kapp.k14s.io/app"] = "app-name"
+	sel := labels.Set(kappLabel).AsSelector()
+
+	resources, err := identifiedResources.List(sel, nil)
+
+	mockedRess.AssertCalled(t, "All")
+	require.Nil(t, err)
+	require.NotNil(t, resources)
+
+	for _, res := range resources {
+		require.Contains(t, res.Labels(), "kapp.k14s.io/app")
+		require.Equal(t, res.Labels()["kapp.k14s.io/app"], kappLabel["kapp.k14s.io/app"])
+	}
+}
+
+type ResourcesMock struct {
+	mock.Mock
+}
+
+func (r *ResourcesMock) All([]ResourceType, AllOpts) ([]Resource, error) {
```

Comment:
instead of having this object be a mock, i would recommend for now to just inline `makeNewResources` functionality into here. it's just a "dummy" implementation of Resources interfaces that always returns 2 resources for this test.